src/Aspire.Dashboard/Extensions/TimeProviderExtensions.cs

```diff
@@ -12,4 +12,21 @@ public static DateTime ToLocal(this TimeProvider timeProvider, DateTimeOffset ut
 
         return dateTime;
     }
+
+    public static DateTime ToLocal(this TimeProvider timeProvider, DateTime dateTime)
+    {
+        if (dateTime.Kind == DateTimeKind.Local)
+        {
+            return dateTime;
+        }
+        if (dateTime.Kind == DateTimeKind.Unspecified)
+        {
+            throw new InvalidOperationException("Unable to convert unspecified DateTime to local time.");
```

Comment:
Can we fall back to some sort of date time kind instead of failing?

Reply:
I don't know what you're suggesting.

This check ensures date times are either explicitly UTC or local. We can't make a good decision if we don't know what kind the date is.